### PR TITLE
Added automatic synchronization of Zabbix hosts after changing DC settings

### LINKED ISSUE #210: Automatically sync Zabbix hosts after changing DC or global MON_ settings
Lets discuss this first...

After changing `MON_ZABBIX_*_VM` DC settings, the user can select all hosts in a DC a press *Update* to force a zabbix sync.

After changing  `MON_ZABBIX_*_NODE` global settings the situation is different and triggering a zabbix sync is not that straightforward.

1) Maybe we should trigger a zabbix sync after changing any of the zabbix settings.
2) Or we should implement an *Update* button for compute nodes.

### PR DESCRIPTION
**Pull request by dn0**
Closes #210

### CODE REVIEW

File: api/dc/base/dc_settings.py [hunk 1]
```diff
@@ -45,18 +44,7 @@ def put(self):
         dc.custom_settings = dcs
         dc.save()
         data = ser.data
```
Comment (secult) [inline api/dc/base/dc_settings.py]: you don't have to reassign this variable to a new name

File: api/mon/base/tasks.py [hunk 2]
```diff
@@ -38,6 +35,37 @@ def mon_clear_zabbix_cache(task_id, dc_id, full=True):
...
+    Clear Zabbix cache and sync everything in Zabbix.
+    Related to a specific DC.
+    Triggered by dc_settings_changed signal.
+    """
+    dc = Dc.objects.get_by_id(int(dc_id))
+
+    if clear_cache:
+        logger.info('Clearing zabbix cache in DC %s', dc)
+        mon_clear_zabbix_cache(dc)
+        get_monitoring(dc)  # Cache new Zabbix instance for tasks below
```
Comment (secult) [inline api/mon/base/tasks.py]: Is this caching useful as the other tasks are run separately?

Comment (dn0) [reply api/mon/base/tasks.py]: @secult Yes it is ... you save calls to the Zabbix API -> look for the `@cache_result@` decorator in api.mon.backends.zabbix. We cache host ids, hostgroup ids, ...

But after changing some monitoring settings we need to clear the cache, e.g. if you change your zabbix server in the settings the IDs are going to change too.

File: api/dc/base/dc_settings.py (hunk 1 above)
Comment (dn0) [reply api/dc/base/dc_settings.py]: You're right, I don't have to assign it. But I have to call `ser.data` before calling `ser.detail_dict()`.
I've added a comment.

File: api/dc/base/dc_settings.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def put(self):
         old_settings = dc.custom_settings
         dc.custom_settings = dcs
         dc.save()
-        data = ser.data
+        data = ser.data  # Prepare ser._data for ser.detail_dict() to work
```
Comment (secult) [inline api/dc/base/dc_settings.py]: In that case, this is very unintuitive and I would suggest removing this dependency, perhaps by changing api.dc.base.serializers.DcSettingsSerializer#detail_dict to prepare the self._data variable if they doesn't exist. Simple accessing a property should not alter the behavior of seemingly non-related method in such a different way.

Comment (secult) [reply api/dc/base/dc_settings.py]: This is Django REST Framework rependent and therefore we don't want to break the compatibility

**Pull request merged**